addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -273,229 +281,241 @@ export class LinkPlugin extends Plugin {
         };
         return pasteAsURLCommand;
     }
+
     /**
-     * Toggle the Link popover to edit links
+     * open the Link popover to edit links
      *
-     * @param {Object} options
-     * @param {HTMLElement} options.link
+     * @param {HTMLElement} [linkElement]
      */
-    toggleLinkTools({ link } = {}) {
-        if (!link) {
-            link = this.getOrCreateLink();
+    openLinkTools(linkElement) {
+        this.closeLinkTools();
+        let selection = this.dependencies.selection.getEditableSelection();
+        let cursorsToRestore = this.dependencies.selection.preserveSelection();
+        const commonAncestor = closestElement(selection.commonAncestorContainer);
+        linkElement = linkElement || findInSelection(selection, "a");
+        if (linkElement) {
+            const linkWasExtended = this.extendsLinkToSelection(linkElement, selection);
```

Comment:
I suggest doing the check here (if it should be extended), and then call the function that does it (extend the link to selection).